java-client/barrage/src/main/java/io/deephaven/client/impl/BarrageSessionFactoryConfig.java

```diff
@@ -4,34 +4,65 @@
 package io.deephaven.client.impl;
 
 import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.barrage.flatbuf.BarrageMessageWrapper;
+import io.deephaven.client.grpc.UserAgentUtility;
 import io.grpc.ManagedChannel;
 import org.apache.arrow.memory.BufferAllocator;
 import org.immutables.value.Value.Default;
 import org.immutables.value.Value.Immutable;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Objects;
 import java.util.concurrent.ScheduledExecutorService;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 @Immutable
 @BuildableStyle
 public abstract class BarrageSessionFactoryConfig {
     private static final SessionConfig SESSION_CONFIG_EMPTY = SessionConfig.builder().build();
 
+    static final List<String> VERSION_PROPERTIES = Collections.unmodifiableList(Stream.concat(
+            FlightSessionFactoryConfig.VERSION_PROPERTIES.stream(),
+            Stream.of(UserAgentUtility.versionProperty("barrage", BarrageMessageWrapper.class)))
+            .collect(Collectors.toList()));
+
+    private static final ClientChannelFactory CLIENT_CHANNEL_FACTORY = ClientChannelFactoryDefaulter.builder()
+            .userAgent(userAgent(Collections.singletonList("deephaven-java-client-barrage")))
+            .build();
+
     public static Builder builder() {
         return ImmutableBarrageSessionFactoryConfig.builder();
     }
 
+    /**
+     * Constructs a <a href="https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#user-agents">grpc
+     * user-agent</a> with {@code grpc-java}, {@code deephaven}, {@code flight}, and {@code barrage} versions, with the
+     * addition of {@code extraProperties}.
+     *
+     * @param extraProperties thee extra properties
+     * @return the user-agent
+     * @see UserAgentUtility#userAgent(List)
+     */
+    public static String userAgent(List<String> extraProperties) {
+        return UserAgentUtility.userAgent(Stream.concat(
+                VERSION_PROPERTIES.stream(),
+                extraProperties.stream())
+                .collect(Collectors.toList()));
+    }
+
     /**
      * The client configuration.
      */
     public abstract ClientConfig clientConfig();
 
     /**
-     * The client channel factory. By default is {@link ClientChannelFactory#defaultInstance()}.
+     * The client channel factory. By default, is a factory that sets a user-agent based on {@link #userAgent(List)}.
```

Comment:
"...sets a user-agent based on #userAgent(List)" - seems like it would help to clarify what that means, potentially at the cost of repeating yourself, rather than jumping back and forth as the reader looks at this?

Reply:
I do struggle a bit, trying to minimize repeating myself (which introduces the possibility of unintentional drift in the future). Let me know what you think about the updated docs.